src/Aspire.Cli/Interaction/ExtensionLoggerProvider.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Cli.Interaction;
+
+internal class ExtensionLoggerProvider(IServiceProvider serviceProvider) : ILoggerProvider
+{
+    public ILogger CreateLogger(string categoryName)
+    {
+        return new ExtensionLogger(serviceProvider);
+    }
+
+    public void Dispose()
+    {
+    }
+}
+
+internal class ExtensionLogger(IServiceProvider serviceProvider) : ILogger
+{
+    private ExtensionInteractionService InteractionService => (ExtensionInteractionService)serviceProvider.GetRequiredService<IInteractionService>();
```

Comment:
You could put `IInteractionService` as a ctor parameter to `ExtensionLoggerProvider` and then pass it through to the ExtensionLogger instead of using IServiceProvider directly.

Reply:
That's what I originally was doing, but it breaks DI since ExtensionBackchannel (used in ExtensionInteractionService) has an ILogger ctor param and so there's an init loop